x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -211,11 +245,165 @@ export const AlertsTableComponent: React.FC<AlertsTableComponentProps> = ({
     [addToCaseBulkActions, addBulkToTimelineAction]
   );
 
-  if (loading || isEmpty(selectedPatterns)) {
+  const { deleteQuery, setQuery } = useGlobalTime(false);
+  // create a unique, but stable (across re-renders) query id
+  const uniqueQueryId = useMemo(() => `${ALERTS_GROUPING_ID}-${uuidv4()}`, []);
+
+  const additionalFilters = useMemo(() => {
+    try {
+      return [
+        buildEsQuery(undefined, globalQuery != null ? [globalQuery] : [], [
+          ...(globalFilters?.filter((f) => f.meta.disabled === false) ?? []),
+          ...(defaultFiltersMemo ?? []),
+        ]),
+      ];
+    } catch (e) {
+      return [];
+    }
+  }, [defaultFiltersMemo, globalFilters, globalQuery]);
+
+  const [groupsActivePage, setGroupsActivePage] = useState<number>(0);
+  const [groupsItemsPerPage, setGroupsItemsPerPage] = useState<number>(25);
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex: groupsActivePage,
+      pageSize: groupsItemsPerPage,
+      onChangeItemsPerPage: (itemsPerPageNumber: number) =>
+        setGroupsItemsPerPage(itemsPerPageNumber),
+      onChangePage: (pageNumber: number) => setGroupsActivePage(pageNumber),
+    }),
+    [groupsActivePage, groupsItemsPerPage]
+  );
+
+  const queryGroups = useMemo(
+    () =>
+      getAlertsGroupingQuery({
+        additionalFilters,
+        selectedGroup,
+        from,
+        runtimeMappings,
+        to,
+        pageSize: pagination.pageSize,
+        pageIndex: pagination.pageIndex,
+      }),
+    [
+      additionalFilters,
+      selectedGroup,
+      from,
+      runtimeMappings,
+      to,
+      pagination.pageSize,
+      pagination.pageIndex,
+    ]
+  );
+
+  const {
+    data: alertsGroupsData,
+    loading: isLoadingGroups,
+    refetch,
+    request,
+    response,
+    setQuery: setAlertsQuery,
+  } = useQueryAlerts<
+    {},
+    GroupingTableAggregation &
+      Record<string, { value?: number | null; buckets?: Array<{ doc_count?: number | null }> }>
+  >({
+    query: queryGroups,
+    indexName: signalIndexName,
+    queryName: ALERTS_QUERY_NAMES.ALERTS_GROUPING,
+    skip: !selectedGroup || selectedGroup === 'none',
```

Comment:
Do you think it may make sense to default `selectedGroup` to 'none' so null or undefined are never actually possible? Just to limit the number of 'empty' states for consistency?